File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
  def zeroFilledSubarray(self, nums: List[int]) -> int:
    nums += [1]
    s = []
    c = 0
    for num in nums:
      if num == 0:
        c += 1
      else:
        if c != 0:
          s.append(c)
          c = 0
    return sum([n*(n+1)//2 for n in s])

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        int n = nums.size();
        int l = 0, r = 0;
        long long ans = 0;
        while (l < n && nums[l] != 0) ++l;
        while (l < n) {
            r = l;
            while (r < n && nums[r] == 0) ++r;
            --r;
            long long len = r-l+1;
            ans += ((len) * (len+1)) / 2;
            l = r+1;
            while (l < n && nums[l] != 0) ++l;
        }
        return ans;
    }
};